vmm/src/device_tree.rs

```diff
@@ -134,6 +134,7 @@ impl<'a> DoubleEndedIterator for BftIter<'a> {
 }
 
 #[cfg(test)]
+#[cfg(feature = "pci_support")]
```

Comment:
Why do you need to make these tests depend on `pci_support`? They are passing just fine for the `mmio` case.

Reply:
Right...yes I agree, the reason was the `DeviceTree` struct implemented the `remove` [method](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/5ebd02a5720a713595c225670f1708d0f0e1191d/vmm/src/device_tree.rs#L65) which needs the `pci_support`. As I think current AArch64 implementation does not use this `device_tree` (for snapshotting), I just skipped unit test of this part.

Reply:
I understand but the `DeviceTree` is used by both `mmio` and `pci`, so I'd rather keep it.

Reply:
Sure, I will try to refactor the code then :)

Reply:
Hi @sboeuf! I picked the testing of the `remove()` case out on AArch64 and did a little bit modification in the `iter()` case. Now the unit test in `device_tree` can work. Please take a review again, thanks!